### DISCUSSION TITLE
Apply matrix-js-sdk

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13663] [Q&A] [answered]
How to use [matrix-js-sdk](https://github.com/matrix-org/matrix-js-sdk) in titanium? Do I have to download the browser version and add it in the same way like d3.js to a webView or add it to the app/lib directory and load it via `require`?

**Comment by m1ga** [comment, marked as answer]
You have to try if it works. Can be either the web version as you've explained or the npm version (https://blog.axway.com/learning-center/software-development/api-development/using-npm-packages-in-titanium). But it depends on the core packages that the module might require if it works or not.

The solution that works but involves a lot more development: use the native SDKs and create a module. E.g. with https://github.com/matrix-org/matrix-android-sdk2 for Android.

**Reply by beppo-ivel** [reply]
The [link](https://titaniumsdk.com/guide/Titanium_SDK/Titanium_SDK_How-tos/Extending_Titanium_Mobile/Android_Module_Development_Guide/Android_Module_Quick_Start.html#undefined) for [Installing Required Python Packages](https://titaniumsdk.com/guide/Titanium_SDK/Titanium_SDK_How-tos/Extending_Titanium_Mobile/Android_Module_Development_Guide/Android_Module_Quick_Start.html#undefined) is undefined. Which are the required packages, only setuptools and markdown or something else?

**Reply by m1ga** [reply]
ah, you can ignore that. That is some legacy info for very old modules. Think I'll remove that.

**Reply by beppo-ivel** [reply]
> You have to try if it works. Can be either the web version as you've explained or the npm version (https://blog.axway.com/learning-center/software-development/api-development/using-npm-packages-in-titanium). But it depends on the core packages that the module might require if it works or not.

The js runtime is [V8 for android](https://titaniumsdk.com/guide/Titanium_SDK/Titanium_SDK_Getting_Started/Titanium_Platform_Overview.html#titanium-at-10-000-feet) and matrix-js-sdk depends on [crypto.ts](https://github.com/matrix-org/matrix-js-sdk/tree/develop/src/crypto). It seems that [crypto module](https://github.com/NativeScript/NativeScript/issues/8806) is not supported by V8. When this issue going to be solved, then matrix-js-sdk will work? Or is it possible when V8 becomes support for crypto, then Apple or Google introduce changes to their respective operating systems that break the titanium sdk version and then a titanium app is not executable at all? Till now it seems matrix-js-sdk wont work in other plattforms like react-native as well. Because the major design patterns are [quite similar](https://stackoverflow.com/questions/36799730/main-differences-between-appcelerator-titanium-and-react-native). What is your guess, will titaniums sdk still alive in future or do you switch to other platforms likenativescript or react-native?

**Reply by m1ga** [reply]
For your last question: it's fully open source now so it won't go away. It still supports the latest mobile SDKs so you can build apps for the store or update existing apps. There are no announcements by Google/Apple that would introduce a breaking change for Ti that stops it from creating apps in the next year (or two, or three). Ti 12 beta is about the be releaed in the next few days. Support for Material 3 is available in the core (it is already possible if you use it by hand), many bugfixes and new features (check my [Titanium News](https://fromzerotoapp.com/titanium-news/) articles). So personally I won't switch to any other platform and still use Ti.

For the thread question: as mentioned before, you can use their native SDK and build a module. But you'll need Java and ObjectivC/Swift knowledge to do so. Starter tutorial for Android: https://fromzerotoapp.com/create-android-modules-for-titanium/ 
You shouldn't have any limitation if you use that
